arch/arm/core/aarch32/cortex_m/mpu/arm_core_mpu.c

```diff
@@ -92,6 +92,18 @@ static const struct z_arm_mpu_partition static_regions[] = {
 			.attr = K_MEM_PARTITION_P_RO_U_NA,
 		},
 #endif /* !CONFIG_MULTITHREADING && CONFIG_MPU_STACK_GUARD */
+#if defined(CONFIG_ETH_STM32_HAL_USE_SRAM3)
```

Comment:
This section is 
- SoC specific, not feature specific, and,
- outside the image boundaries (image_ram_start, image_ram_end)

So I think we should try to define this MPU region at boot. See soc/arm/common/arm_mpu_regions.c. As discussed in the dev-review meeting there is a Kconfig option that allows to override the default SoC MPU region definition, and add a SoC-specific one that includes _this_ region as well.

Reply:
Thanks for this hint! Added SoC specific MPU region definitions.